### PR TITLE
move genPVC out of api

### PR DESCRIPTION
**Pull request by yashbhutwala**
/assign @rrati

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/714#" title="Author self-approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/utils/common.go [hunk 1]
```diff
@@ -54,3 +58,82 @@ func SetLimits(containerConfig *api.ContainerConfig, config types.Container) {
 		containerConfig.MaxMem = fmt.Sprintf("%dM", *config.MaxMem)
 	}
 }
+
+func GenPVC(b blackduckapi.Blackduck, defaultPVC map[string]string) ([]*components.PersistentVolumeClaim, error) {
```
Comment (rrati) [inline pkg/apps/utils/common.go]: This is a black duck specific function.  I would make it attached to a blackduck related entity under pkg/apps/blackduck.  IMO, creating a base struct with
 	config     *protoform.Config
	kubeClient *kubernetes.Clientset
	blackduck  *blackduckapi.Blackduck

and get GenPVC function, and have that embedded into the v[1|2]/BdPVC struct.

File: pkg/apps/blackduck/blackduck.go [hunk 2]
```diff
@@ -2128,32 +2128,80 @@ func (b Blackduck) isBinaryAnalysisEnabled(bdspec *blackduckapi.BlackduckSpec) b
 	return false
 }
 
-func (b Blackduck) createPVC(name string, requestedSize string, defaultSize string, storageclass string, accessMode horizonapi.PVCAccessModeType, label map[string]string, blackduck *blackduckapi.Blackduck) *components.PersistentVolumeClaim {
+func GenPVC(blackDuck blackduckapi.Blackduck, defaultPVC map[string]string) ([]*components.PersistentVolumeClaim, error) {
+	var pvcs []*components.PersistentVolumeClaim
```
Comment (msenmurugan) [inline pkg/apps/blackduck/blackduck.go]: @rrati @yashbhutwala not part of this PR.. Probably we should create a common method to support for all apps. We need to make all the app consistent to get PVC storage class, claim size of each PVC, volume name associating to each PVC. so that the customer will not be confused. I think, it is easy to be done.

Comment (rrati) [reply pkg/apps/blackduck/blackduck.go]: A common means to define/generate PVCs is a good path.  We'll need that as we add more apps to the operator.

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/714#pullrequestreview-269156584" title="LGTM">rrati</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/714#" title="Author self-approved">yashbhutwala</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [rrati,yashbhutwala]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
